### ISSUE TITLE
jsdom tests

### ISSUE THREAD

**Issue by Thumster** [dequelabs/axe-core#4177]
### Product

axe-core

### Question

version of axe-core: v4.8.1

I've used the jsdom example provided as a starting guide. I'm trying to better understand the incomplete scenarios and thus trying to recreate the scenarios with jsdom to trigger the respective incomplete messages. However, I am unable to do so for some tests.

e.g. For the `area-alt` rule. I'm trying to trigger the `aria-labelledby` check that potentially returns an incomplete message `ensure aria-labelledby references an existing element`. However, the incomplete message never appears in the run, I've taken a look at the tests and this scenario is only triggered when using the `SerialVirtualNode` is in play. Is there any way to recreate this in jsdom?

html example
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <title>JSDOM Example</title>
</head>

<body>
  <map id="incomplete-area-alt">
    <area href="foobar" aria-labelledby="foobar" />
  </map>
</body>

</html>
```

Result:
```js
{
  violations: 1,
  incomplete: 0
}
```

Expected result:
similar to `should incomplete for aria-labelledby` test case in `virual-rules`
```js
{
  violations: 0,
  incomplete: 1
}
```

**Comment by WilcoFiers** [comment]
@Thumster First things first. Can you explain why you are doing this? That would help me answer your question better.

**Comment by Thumster** [comment]
Hi @WilcoFiers , just to give some context, I'm looking to include needs review items (incomplete) in my scans. However, I want to get a clearer understanding of the incomplete findings before I do so. My application is making use of Crawlee and running `axe.run()` in `page.evaluate`.

To try and recreate the incomplete findings, I'm making use of jsdom to do so.

**Comment by WilcoFiers** [comment]
Thanks for clarifying. But why do you want to know how to trigger incompletes? What is the use  case for knowing how to do that? I'm digging a little, because if there's some use case we don't know about that we may want to support I'd like to understand it.

To answer your question; I'm not sure this can be triggers with JSDOM. Maybe with JSDOM Fragement. You'd have to try. We don't keep track of exactly what environments different things can and can't occur. Axe-core triggers incompletes for lots of reasons, from things it can't do anywhere (like test contrast on images), to things that we're not sure can ever really happen (like unsupported colors from getComputedStyle), but we'll often do it just in case. With so many environments this tool is used in we've learned not to assume.

Not sure that's much of an answer for you but I think that's the best I can do.

**Comment by Thumster** [comment]
The main if not only reason is to better understand the scenarios before deciding whether or not to display the incomplete messages. Currently, when we decide to show all the incomplete messages, it gets a little lengthy and we are deciding to hide some of them and hence trying to understand each of them before hiding them.

Thanks for the JSDOM suggestion. I will try that out and give my updates here. Your explanation helps a bunch, thanks @WilcoFiers. Really appreciate all the effort and work that goes into axe-core 🙏
